augeas: Use generic augeas modules

mkutlak opened the pull request:
Most of the changes are just replacement of current regexps with in-build modules
from augeas. The changes do not make any major changes in the behaviour of the lenses.

The only thing that is different is that Util.empty [1] module ignores empty comments.
This means that empty commented lines in the configuration files do not show up in
augeas tree and because of that they can't be changed using the augeas.

The goal of this PR is to simplify the augeas lenses but it should also fix the linked
bugzilla bug.

[1] http://augeas.net/docs/references/1.4.0/lenses/files/util-aug.html#Util.empty

Related: rhbz#1649742

Signed-off-by: Martin Kutlak <mkutlak@redhat.com>

jfilak commented:
OK, and why do you think it is a good idea to remove empty comments?

jfilak commented:
Out of curiosity, why was this PR created?

mkutlak commented:
> OK, and why do you think it is a good idea to remove empty comments?

Maybe I am wrong but I don't think the empty comments are useful in augeas tree.

> Out of curiosity, why was this PR created?

Side-effect of working on BZ#1477476.

jfilak commented:
> Maybe I am wrong but I don't think the empty comments are useful in augeas tree.

It might be a simplification but I would rather not remove anthing user has created - for example empty lines for easier orientation.

> https://bugzilla.redhat.com/show_bug.cgi?id=1477476

So, is the reason to strip double quotes and white spaces from the cofigured values?

mkutlak commented:
> I would rather not remove anthing user has created - for example empty lines for easier orientation.

Everything user has created stays.

Let me demonstrate:
```
$ cat /etc/abrt/abrt-action-save-package-data.conf
# With this option set to "yes",
# only crashes in signed packages will be analyzed.
# the list of public keys used to check the signature is
# in the file gpg_keys
#
# How can I check the GPG key used to sign an installed package on
# Red hat Enterprise Linux:
# https://access.redhat.com/solutions/1120013
#
# Product Signing (GPG) Keys:
# https://access.redhat.com/security/team/key
#
OpenGPGCheck = yes

# Blacklisted packages
...
```
Older version:
```
$ augtool print /files/etc/abrt/abrt-action-save-package-data.conf
/files/etc/abrt/abrt-action-save-package-data.conf
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[1] = "With this option set to \"yes\","
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[2] = "only crashes in signed packages will be analyzed."
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[3] = "the list of public keys used to check the signature is"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[4] = "in the file gpg_keys"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[5] = ""
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[6] = "How can I check the GPG key used to sign an installed package on"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[7] = "Red hat Enterprise Linux:"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[8] = "https://access.redhat.com/solutions/1120013"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[9] = ""
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[10] = "Product Signing (GPG) Keys:"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[11] = "https://access.redhat.com/security/team/key"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[12] = ""
/files/etc/abrt/abrt-action-save-package-data.conf/OpenGPGCheck = "yes"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[13] = "Blacklisted packages"
```
PR version:
```
$ augtool print /files/etc/abrt/abrt-action-save-package-data.conf 
/files/etc/abrt/abrt-action-save-package-data.conf
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[1] = "With this option set to \"yes\","
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[2] = "only crashes in signed packages will be analyzed."
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[3] = "the list of public keys used to check the signature is"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[4] = "in the file gpg_keys"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[5] = "How can I check the GPG key used to sign an installed package on"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[6] = "Red hat Enterprise Linux:"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[7] = "https://access.redhat.com/solutions/1120013"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[8] = "Product Signing (GPG) Keys:"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[9] = "https://access.redhat.com/security/team/key"
/files/etc/abrt/abrt-action-save-package-data.conf/OpenGPGCheck = "yes"
/files/etc/abrt/abrt-action-save-package-data.conf/#comment[10] = "Blacklisted packages"
```

With the PR version empty comments are removed from the augeas tree so you are not able to do:

`augtool set /files/etc/abrt/abrt-action-save-package-data.conf/#comment[12] = "Hello World"`

but you have to do:

`augtool ins #comment before ...` and then `augtool set ...`

> So, is the reason to strip double quotes and white spaces from the cofigured values?

No, double quotes and white spaces are still parsed as regular value.

jfilak commented:
If you don't have the comments in the augeas tree, you don't write them to the configuration file after update, right?

>  No, double quotes and white spaces are still parsed as regular value.

Please, tell me what is the reason for this PR, so we can update the commit message, fix the removed comments and eventually merge the  PR.

mkutlak commented:
- Rebased, updated the commit message and linked rhbz

These changes also require to modifications in the [test_abrt.aug](https://github.com/abrt/abrt/blob/master/augeas/test_abrt.aug) in [abrt/abrt](https://github.com/abrt/abrt/).

I will create a PR for it if this PR is merged.

mkutlak commented:
retest this please

ernestask commented:
Makes sense to me.
